test/integration/endpoints_test.go

```diff
@@ -0,0 +1,447 @@
+package integration
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/modelplex/modelplex/internal/config"
+	"github.com/modelplex/modelplex/internal/server"
+)
+
+// TestIntegration_HTTPEndpoints tests the full HTTP endpoint functionality
+func TestIntegration_HTTPEndpoints(t *testing.T) {
+	if testing.Short() {
+		t.Skip("Skipping integration test in short mode")
+	}
+
+	// Create test config
+	cfg := &config.Config{
+		Providers: []config.Provider{
+			{
+				Name:     "test-openai",
+				Type:     "openai",
+				BaseURL:  "http://localhost:8080",
+				APIKey:   "test-key",
+				Models:   []string{"gpt-4", "gpt-3.5-turbo"},
+				Priority: 1,
+			},
+			{
+				Name:     "test-anthropic",
+				Type:     "anthropic",
+				BaseURL:  "http://localhost:8081",
+				APIKey:   "test-key-2",
+				Models:   []string{"claude-3-sonnet"},
+				Priority: 2,
+			},
+		},
+		MCP: config.MCPConfig{
+			Servers: []config.MCPServer{
+				{Name: "filesystem", Command: "test-fs", Args: []string{"/tmp"}},
+				{Name: "brave-search", Command: "test-search"},
+			},
+		},
+		Server: config.Server{
+			LogLevel:       "info",
+			MaxRequestSize: 1024 * 1024,
+		},
+	}
+
+	// Find an available port
+	listener, err := net.Listen("tcp", "127.0.0.1:0")
+	require.NoError(t, err)
+	port := listener.Addr().(*net.TCPAddr).Port
+	listener.Close()
```

Comment:
This listener-to-port functionality is duplicated a bunch. Make it a utility function.